### ISSUE TITLE
[Bug]: X data not normalized for Gaussian Process / Bayes Search

### ISSUE THREAD

**Issue by luifire** [wandb/wandb#10809]
Thank you very much for WandB Sweeps. I really came to love them.

### Describe the bug

When looking at the code, it looks like we are not normalizing X, see [train_gaussian_process](https://github.com/wandb/sweeps/blob/96dc1c028f5c9e804632790bfb00d901fa9a49d1/src/sweeps/bayes_search.py#L87).
Only Y gets normalized in [fit_normalized_gaussian_process](https://github.com/wandb/sweeps/blob/96dc1c028f5c9e804632790bfb00d901fa9a49d1/src/sweeps/bayes_search.py#L44C18-L44C19).

Since the Gauss Process is using this [sklearn Matérn kernel](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.Matern.html#sklearn.gaussian_process.kernels.Matern)
<img width="900" height="158" alt="Image" src="https://github.com/user-attachments/assets/eeaf3f2f-91ab-483b-a5c8-9260178755bd" />

where $d$ is the Euclidean distance, I would assume that some dimensions of X could easily dominate others, e.g. batch size vs. learning rate.

I presume, that it could be harmful since the covariance for known samples
(Example 1)
_Batch size = 32, learning rate = 1e-3
Batch size = 32, learning rate = 1e-5_

is more similar than for 
(Example 2)
_Batch size = 32, learning rate = 1e-4
Batch size = 34, learning rate = 1e-4_

And so the inference for
_Batch size = 33, learning rate = 1e-4_
has a higher variance in the posterior for Example 2, than for Example 1. Even though it should be the other way around.

But maybe I don't understand some parts of the math here?

**Comment by fmamberti-wandb** [comment]
Hi @luifire , thank you for your report we will review this and will get back to you.

**Comment by fmamberti-wandb** [comment]
Hi @luifire thank you for your patience while we reviewed this. The X data passed to `train_gaussian_process` is already normalized as it's normalized upstream using  [normalize_runs_as_array](https://github.com/wandb/sweeps/blob/96dc1c028f5c9e804632790bfb00d901fa9a49d1/src/sweeps/params.py#L477) called by [_construct_gp_data](https://github.com/wandb/sweeps/blob/96dc1c028f5c9e804632790bfb00d901fa9a49d1/src/sweeps/bayes_search.py#L413) run via [bayes_search_next_run](https://github.com/wandb/sweeps/blob/96dc1c028f5c9e804632790bfb00d901fa9a49d1/src/sweeps/bayes_search.py#L521). Please us know if you have any further questions.
